home/tests/test_content_import_export.py

```diff
@@ -1896,3 +1898,125 @@ def test_list_items(self, impexp: ImportExport) -> None:
+                )
+            ],
+        )
+
+        self_help = PageBuilder.build_cp(
+            parent=ha_menu,
+            slug="self-help",
+            title="self-help",
+            bodies=[
+                WABody(
+                    "self-help",
+                    [
+                        WABlk(
+                            "*Self-help programs*",
+                            buttons=[PageBtn("Import Export", page=imp_exp)],
+                        )
+                    ],
+                )
+            ],
+        )
+
+        rev = self_help.save_revision()
+        rev.publish()
+
+        orig = impexp.get_page_json()
+        impexp.export_reimport()
+        imported = impexp.get_page_json()
+        assert imported == orig
+
+    def test_export_missing_go_to_button(self, impexp: ImportExport) -> None:
+        """
+        If a page has a button go to page that no longer exists, the missing button
+        is skipped during export.
+        """
+        home_page = HomePage.objects.first()
+        main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
+        imp_exp = PageBuilder.build_cpi(home_page, "import-export", "Import Export")
+
+        ha_menu = PageBuilder.build_cp(
+            parent=main_menu,
+            slug="ha-menu",
+            title="HealthAlert menu",
+            bodies=[WABody("HealthAlert menu", [WABlk("*Welcome to HealthAlert*")])],
+        )
+        PageBuilder.build_cp(
+            parent=ha_menu,
+            slug="health-info",
+            title="health info",
+            bodies=[
+                WABody(
+                    "health info",
+                    [
+                        WABlk(
+                            "*Health information*",
+                            buttons=[NextBtn("Next message button")],
+                        )
+                    ],
+                )
+            ],
+        )
+        PageBuilder.build_cp(
+            parent=ha_menu,
+            slug="self-help",
+            title="self-help",
+            bodies=[
+                WABody(
+                    "self-help",
+                    [
+                        WABlk(
+                            "*Self-help programs*",
+                            buttons=[PageBtn("Import Export", page=imp_exp)],
+                        )
+                    ],
+                )
+            ],
+        )
+
+        orig = impexp.get_page_json()
+
+        orig_btn = list(orig)[4]["fields"]["whatsapp_body"][0]["value"]["buttons"]
+
+        # delete page referenced by other content page
+        imp_exp.delete()
+
+        impexp.export_reimport()
+        imported = impexp.get_page_json()
+        imported_btn = list(imported)[3]["fields"]["whatsapp_body"][0]["value"][
+            "buttons"
+        ]
+
+        assert imported_btn == []
+        assert orig_btn == [
+            {"type": "go_to_page", "value": {"page": 1, "title": "Import Export"}}
+        ]
```

Comment:
We need to assert on the rest of the content as well, otherwise there could be other breakages that we don't pick up in the test.